### PR TITLE
feat(editable-layers): use the geojson types instead of custom types

### PR DESCRIPTION
**Pull request by giangbv1984**
- Use the types of geojson instead of custom types
- Fixed bug: missed the boxicons
image

### CODE REVIEW

File: examples/editable-layers/editor/package.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "name": "@deck.gl-community/editor",
```
Comment (ibgreen) [inline examples/editable-layers/editor/package.json]: While it wouldn't really matter here, lets only use the npm organization scope for actual modules
```suggestion
  "name": "editable-layers/editor",
```

File: modules/editable-layers/src/utils/geojson-types.ts [hunk 2]
```diff
@@ -4,7 +4,17 @@
 
 // Geometry types
 
-export type Position = [number, number] | [number, number, number];
+import {
```
Comment (ibgreen) [inline modules/editable-layers/src/utils/geojson-types.ts]: Any reason why we need to alias names? Can't we just re-export "as is" from the geojson module?

File: modules/editable-layers/src/utils/geojson-types.ts [hunk 3]
```diff
@@ -4,7 +4,17 @@
...
+  Point as PointGeo,
+  LineString as LineStringGeo,
+  Polygon as PolygonGeo,
+  MultiPoint as MultiPointGeo,
+  MultiLineString as MultiLineStringGeo,
+  MultiPolygon as MultiPolygonGeo,
+  Position as PositionGeo,
+} from 'geojson';
```
Comment (ibgreen) [inline modules/editable-layers/src/utils/geojson-types.ts]:
```suggestion
export {
  Point,
  LineString,
  Polygon,
  MultiPoint,
  MultiLineString,
  MultiPolygon,
  Position
} from 'geojson';
```

**Review by ibgreen — COMMENTED** [review]
I am temporarily confused. Looking at npm, the geojson package doesn't seem to be exporting any types?
- https://www.npmjs.com/package/geojson
- https://github.com/caseycesari/GeoJSON.js/blob/master/geojson.js

File: modules/editable-layers/src/utils/geojson-types.ts (hunk 3 above)
Comment (giangbv1984) [reply modules/editable-layers/src/utils/geojson-types.ts]: I have resolved it.

File: examples/editable-layers/editor/package.json (hunk 1 above)
Comment (giangbv1984) [reply examples/editable-layers/editor/package.json]: Reverted file

**Comment by giangbv1984** [comment]
Hi @ibgreen. Can you help me to review this PR? Thanks!

File: modules/editable-layers/src/utils/geojson-types.ts [hunk 4]
```diff
@@ -4,7 +4,26 @@
...
+
+export {
+  Point,
+  LineString,
+  Polygon,
+  MultiPoint,
+  MultiLineString,
+  MultiPolygon,
+  Position
+};
```
Comment (ibgreen) [inline modules/editable-layers/src/utils/geojson-types.ts]:
```suggestion
} from 'geojson';
```

File: modules/editable-layers/src/utils/geojson-types.ts [hunk 5]
```diff
@@ -20,36 +39,6 @@ export type AnyCoordinates =
...
-  type: 'MultiLineString';
-  coordinates: MultiLineStringCoordinates;
-};
-
-export type MultiPolygon = {
-  type: 'MultiPolygon';
-  coordinates: MultiPolygonCoordinates;
-};
-
 export type Geometry = Point | LineString | Polygon | MultiPoint | MultiLineString | MultiPolygon;
```
Comment (ibgreen) [inline modules/editable-layers/src/utils/geojson-types.ts:34]: Unrelated to this PR
- Add a `/** */` TSDoc style comment for each new definition. 
- I am not sure I like the `Geometry` and `Polygonal` types as these are not following normal conventions. But changing them probably requires updating other code, so would be for a different PR.
- For one thing, `Geometry` is typically called `Feature`

File: modules/react/package.json [hunk 6]
```diff
@@ -40,7 +40,8 @@
     "prop-types": "^15.8.1",
     "styled-components": "^4.4.1",
     "styled-react-modal": "^3.1.1",
-    "supercluster": "^8.0.1"
+    "supercluster": "^8.0.1",
+    "boxicons": "^2.1.4"
```
Comment (ibgreen) [inline modules/react/package.json:44]: Why are we adding `boxicons`? Was it missing? Kind of unrelated to this PR...

**Review by ibgreen — APPROVED** [review]

File: modules/react/package.json (hunk 6 above)
Comment (giangbv1984) [reply modules/react/package.json:44]: Because, the modules/react/src/components/icon.tsx use `boxicons `. If we don't install this library, we have an error (You can see the description) when running test

File: modules/editable-layers/src/utils/geojson-types.ts (hunk 5 above)
Comment (giangbv1984) [reply modules/editable-layers/src/utils/geojson-types.ts:34]: This is your code. I only import the types of geojson.

**Comment by ibgreen** [comment]
@giangbv1984 this is approved. If you can make it pass the linter (run `yarn lint fix` and push again) I will land this.

**Comment by giangbv1984** [comment]
> @giangbv1984 this is approved. If you can make it pass the linter (run `yarn lint fix` and push again) I will land this.

I have runned `yarn lint fix` and fixed the import @types/geojson. But, we still have a lot of warning:
image

**Comment by ibgreen** [comment]
Had to land this manually to fix the linter issue.
